Nginx docker improvements

kwypchlo opened the pull request:
- use docker-entrypoint script from official docker-nginx repo to improve task management
  - replace manual templating with envsubst entrypoint script from official docker-nginx repo
    - it doesn't support conditionals but I found out that even if we have 2 nginx servers with the same name in configuration, it will properly skip the latter which is what we actually did with the conditions
  - replace inline CMD script for reloading nginx config every 6 hours (for letsencrypt) with entrypoint script (copied from pull request on docker-nginx repo)
  - replace build step that produces local ssl certificate with entrypoint script that does exactly that - we probably don't want to include this in build step since it will end up in the image when we build it and deploy on dockerhub (probably bad practice)
- enable pcre_jit to speed up regex processing (copied from openresty nginx defaults)
- replaced copying over nginx configuration in dockerfile to mounting it with volumes
- use [skynetlabs/nginx](https://hub.docker.com/r/skynetlabs/nginx) docker image from dockerhub
- add docker/nginx/testing/Dockerfile to hadolint

peterjan commented on docker/nginx/Dockerfile:
```diff
@@ -2,25 +2,20 @@ FROM openresty/openresty:1.19.9.1-focal
-COPY conf.d /etc/nginx/conf.d
-COPY conf.d.templates /etc/nginx/conf.d.templates
-COPY nginx.conf /usr/local/openresty/nginx/conf/nginx.conf
+# reload nginx every 6 hours (for reloading certificates)
```
🕺  woop woop - nice

kwypchlo commented on docker/nginx/conf.d.templates/server.hns.conf.template:
```diff
@@ -0,0 +1,41 @@
+server {
+	server_name *.hns.${SERVER_DOMAIN}; # example: *.hns.eu-ger-1.siasky.net
+	
+	set_by_lua_block $skynet_portal_domain { return "${SERVER_DOMAIN}" }
```
yeah, I'll add a comment

ro-tex reviewed (dismissed):
LGTM.

Parts of the new code are quite hard to review, so I basically trust that they behave in the expected way and that this has been manually tested to give the right results.

kwypchlo commented:
> LGTM.
> 
> Parts of the new code are quite hard to review, so I basically trust that they behave in the expected way and that this has been manually tested to give the right results.

I renamed the files and deleted some conditions (deployed and tested) that's why they appear new
